compiler/rustc_traits/src/normalize_erasing_regions.rs

```diff
@@ -57,6 +57,38 @@ fn try_normalize_after_erasing_regions<'tcx, T: TypeFoldable<'tcx> + PartialEq +
     })
 }
 
+#[instrument(level = "debug", skip(tcx))]
+fn try_normalize_for_mir_after_erasing_regions<'tcx, T: TypeFoldable<'tcx> + PartialEq + Copy>(
```

Comment:
These two functions are the same except for the normalization method they use. Maybe pass that method as a param and only have one copy of the function?